File: ActivityLifecycle/app/src/main/java/com/example/activitylifecycle/MainActivity.java
package com.example.myapplication;

import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.widget.Toast;


public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Log.d("Lifecycle", "onCreate called");
        Toast.makeText(this, "onCreate", Toast.LENGTH_SHORT).show();
    }

    @Override
    protected void onStart() {
        super.onStart();
        Log.d("Lifecycle", "onStart called");
        Toast.makeText(this, "onStart", Toast.LENGTH_SHORT).show();
    }

    @Override
    protected void onResume() {
        super.onResume();
        Log.d("Lifecycle", "onResume called");
        Toast.makeText(this, "onResume", Toast.LENGTH_SHORT).show();
    }

    @Override
    protected void onPause() {
        super.onPause();
        Log.d("Lifecycle", "onPause called");
        Toast.makeText(this, "onPause", Toast.LENGTH_SHORT).show();
    }

    @Override
    protected void onStop() {
        super.onStop();
        Log.d("Lifecycle", "onStop called");
        Toast.makeText(this, "onStop", Toast.LENGTH_SHORT).show();
    }

    @Override
    protected void onRestart() {
        super.onRestart();
        Log.d("Lifecycle", "onRestart called");
        Toast.makeText(this, "onRestart", Toast.LENGTH_SHORT).show();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        Log.d("Lifecycle", "onDestroy called");
        Toast.makeText(this, "onDestroy", Toast.LENGTH_SHORT).show();
    }
}

File: Activity/app/src/main/java/com/example/activity/MainActivity.kt
package com.example.activity

import android.os.Bundle
import android.util.Log
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    protected override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        print("lifecycle onCreate invoked")
    }

    protected override fun onStart() {
        super.onStart()
        Log.d("lifecycle", "onStart invoked")
        Toast.makeText(getApplicationContext(), "onStart Called", Toast.LENGTH_LONG).show()
    }

    protected override fun onResume() {
        super.onResume()
        Log.d("lifecycle", "onResume invoked")
        Toast.makeText(getApplicationContext(), "onResume Called", Toast.LENGTH_LONG).show()
    }

    protected override fun onPause() {
        super.onPause()
        Log.d("lifecycle", "onPause invoked")
        Toast.makeText(getApplicationContext(), "onPause Called", Toast.LENGTH_LONG).show()
    }

    protected override fun onStop() {
        super.onStop()
        Log.d("lifecycle", "onStop invoked")
        Toast.makeText(getApplicationContext(), "onStop Called", Toast.LENGTH_LONG).show()
    }

    protected override fun onRestart() {
        super.onRestart()
        Log.d("lifecycle", "onRestart invoked")
        Toast.makeText(getApplicationContext(), "onRestart Called", Toast.LENGTH_LONG).show()
    }

    protected override fun onDestroy() {
        super.onDestroy()
        Log.d("lifecycle", "onDestroy invoked")
        Toast.makeText(getApplicationContext(), "onDestroy Called", Toast.LENGTH_LONG).show()
    }
}